server.js

```diff
@@ -0,0 +1,87 @@
+/**
+ *
+ * This application transforms socket messages into MQTT messages
+ *
+ * @author  Dennis de Greef <github@link0.net>
+ * @license MIT
+ *
+ * Requires JSON structure for each request according to the following format (on a single line)
+ *
+ * {
+ *     secret: "youshouldchangethis",
+ *     topic: "/user/example/foo",
+ *     message: "This is a message for the queue"
+ * }
+ *
+ * in which you'll get a response with a boolean whether the request was accepted
+ *
+ * {
+ *     success: true
+ * }
+ *
+ */
+var mqtt     = require('mqtt');
+var net      = require('net');
+var config   = require('./config');
+
+var mqttUri  = 'mqtt://' + config.mqtt.hostname + ':' + config.mqtt.port;
+var client   = mqtt.connect(mqttUri);
+
+client.on('connect', function () {
+    log("Connected to MQTT: " + mqttUri);
+    log("Starting to listen on tcp://" + config.socket.host + ':' + config.socket.port);
+    net.createServer(function(sock) {
+        sock.on('data', function(streamData) {
+            var data = streamData.toString();
+
+            var json = validateJSON(data.replace(/^\s+|\s+$/g, ''));
+            if (json && json.secret == config.socket.secret) {
```

Comment:
Whenever I'm developing in another, mostly object oriented, language, I always adhere to the "Only one indentation level per method"-rule (https://speakerdeck.com/rdohms/your-code-sucks-lets-fix-it?slide=21). Somehow, whenever I'm doing Javascript, that rule gets throw out the window.

The one thing that might be an improvement, is switching the if-else-statement and use an early return.
Whilst you're at it, you might want to create different failures for the secret and the json. Although that has an impact on security, it might be a lot easier with debugging.

What do yo think?